feat: pass markdown for hint message on field

gdgkirkley opened the pull request:
**Summary**
This adds the option to include HTML in the hint message on a field. Closes #5559 

The issue creator requested support for Markdown, which could be added in future but would require parsing the markdown from the config and would be a larger change. This is a simpler addition that addresses the same issue.

**Test plan**
I didn't see any existing tests for these components, but would be more than happy to add if I just missed them! This passes the existing tests.

![image](https://user-images.githubusercontent.com/44145447/124407083-e744a300-dcf7-11eb-96af-bf3a373d99b4.png)

**A picture of a cute animal (not mandatory but encouraged)**
![Cute pup](https://pbs.twimg.com/media/E5TrCL5X0AU1QLT?format=jpg&name=small)

erezrokah commented on dev-test/config.yml:
```diff
@@ -50,7 +50,7 @@ collections: # A list of collections the CMS should be able to edit
         tagname: ''
 
-      - { label: 'Body', name: 'body', widget: 'markdown', hint: 'Main content goes here.' }
+      - { label: 'Body', name: 'body', widget: 'markdown', hint: 'Main content goes here. <a href="http://example.com">Can include HTML!</a>' }
```
Can we revert this change?

gdgkirkley replied:
Yep!

erezrokah commented on packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:
```diff
@@ -327,9 +331,11 @@ class EditorControl extends React.Component {
-              <ControlHint active={isSelected || this.state.styleActive} error={hasErrors}>
-                {fieldHint}
-              </ControlHint>
+              <ControlHint
```
I believe the intention in the issue is to support markdown, not injecting HTML directly (which we would like to avoid).

We could use https://github.com/remarkjs/react-markdown or even [expose the one we're using for the markdown widget preview](https://github.com/netlify/netlify-cms/blob/574025fd649d49f70650a6112617642a6cb24f07/packages/netlify-cms-widget-markdown/src/MarkdownPreview.js#L20)

gdgkirkley replied:
Ah cool - sounds good! I will update to use one or the other with markdown.

erezrokah reviewed (changes requested):
Thanks @gdgkirkley, can you look into supporting markdown and not HTML directly?

gdgkirkley commented:
Hey @erezrokah - I've updated to use `react-markdown` and removed that change to the dev config. `react-markdown` seemed like a great option since it can limit the allowed elements to just the bold, itatlic and link tags requested in the issue.

I also made an update to the docs for this change - hope that makes sense for this change!

erezrokah commented on packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:
```diff
@@ -328,7 +329,23 @@ class EditorControl extends React.Component {
               <ControlHint active={isSelected || this.state.styleActive} error={hasErrors}>
-                {fieldHint}
+                <ReactMarkdown
+                  allowedElements={['a', 'strong', 'em']}
```
`em` should be used to support italic.

Also, with `unwrapDisallowed=true` we don't need the `p` element or the added styling.

erezrokah commented on packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:
```diff
@@ -328,7 +329,23 @@ class EditorControl extends React.Component {
+                    a: ({ node, ...props }) => (
+                      <a
+                        {...props}
+                        target="_blank"
```
Open links in a new tab

erezrokah commented on packages/netlify-cms-core/src/components/Editor/EditorControlPane/EditorControl.js:
```diff
@@ -328,7 +329,23 @@ class EditorControl extends React.Component {
+                        {...props}
+                        target="_blank"
+                        rel="noopener noreferrer"
+                        style={{ color: 'inherit' }}
```
Inherit the style for links

erezrokah reviewed (approved):
Thanks for the follow up @gdgkirkley!

I've made some minor changes to fix italic, open links in a new tabs and also apply the active/error style (active - when the field is focused, error when for example the field is required and the user clicks save).

![image](https://user-images.githubusercontent.com/26760571/124741382-4d157280-df24-11eb-98f4-f9aae2a1ac55.png)

![image](https://user-images.githubusercontent.com/26760571/124741407-543c8080-df24-11eb-9e3e-e85e87f4b671.png)
